protocol/message.go

```diff
@@ -39,10 +57,39 @@ type DirectoryProof struct {
 	AP    *m.AuthenticationPath
 	STR   *m.SignedTreeRoot
 	TB    *m.TemporaryBinding `json:",omitempty"`
-	Error ErrorCode           `json:",omitempty"`
+	Error ErrorCode
+}
+
+type DirectoryProofInEpoch struct {
+	Type  int
+	AP    *m.AuthenticationPath
+	STR   []*m.SignedTreeRoot
+	Error ErrorCode
+}
+
+type DirectoryProofs struct {
```

Comment:
>  How would a client that doesn't know how many epochs have passed use the Limit?

Hmm, I think the client should lookup for range from lookup epoch to its current latest epoch (the epoch of current saved STR). Even so, the `Limit` field seems to make no sense here. I originally thought that if the client want to do a key lookup in epoch, say, a month ago, then I'm not sure how big the response's size is?

Let me think more about it ...